callback_var.go

```diff
@@ -0,0 +1,63 @@
+package goflags
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// CallBackFunc
+type CallBackFunc func()
+
+// callBackVar
+type callBackVar struct {
+	Value CallBackFunc
+}
+
+// Set
+func (c *callBackVar) Set(s string) error {
+	v, err := strconv.ParseBool(s)
+	if err != nil {
+		return fmt.Errorf("failed to parse callback flag")
+	}
+	if v {
+		// if flag found execute callback
+		c.Value()
+	}
+	return nil
+}
+
+// IsBoolFlag
+func (c *callBackVar) IsBoolFlag() bool {
```

Comment:
Is this used anywhere?

Reply:
@Mzack9999 this is method required for implementing custom boolean flag .

Reply:
This will be used by the actual `flagSet.CommandLine.Parse(os.Args[1:])`., just to know that flag is of type boolean

flag.go
```go
// optional interface to indicate boolean flags that can be
// supplied without "=value" text
type boolFlag interface {
	Value
	IsBoolFlag() bool
}
```
usage snippet:
```go
....
if fv, ok := flag.Value.(boolFlag); ok && fv.IsBoolFlag() { // special case: doesn't need an arg
		if hasValue {
			if err := fv.Set(value); err != nil {
				return false, f.failf("invalid boolean value %q for -%s: %v", value, name, err)
			}
		}
		...
```

Reply:
Clear, thanks for clarifying!